src/Sylius/Bundle/ResourceBundle/Twig/ResourceExtension.php

```diff
@@ -52,20 +53,36 @@ class ResourceExtension extends \Twig_Extension
      */
     private $router;
 
+    /** @var CsrfProviderInterface */
+    protected $csrfProvider;
+
+    /** @var string */
+    protected $intention;
+
     /**
      * Constructor.
      *
-     * @param RouterInterface $router
-     * @param Parameters      $parameters
-     * @param string          $paginateTemplate
-     * @param string          $sortingTemplate
+     * @param RouterInterface       $router
+     * @param Parameters            $parameters
+     * @param string                $paginateTemplate
+     * @param string                $sortingTemplate
+     * @param CsrfProviderInterface $csrfProvider
+     * @param string                $intention
      */
-    public function __construct(RouterInterface $router, Parameters $parameters, $paginateTemplate, $sortingTemplate)
```

Comment:
Humm This class become fat and I don't like it. I think you should create a new extension, what do you think ?